airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessBuilderFactory.java

```diff
@@ -80,4 +86,18 @@ private Path rebasePath(final Path jobRoot) {
     return DATA_MOUNT_DESTINATION.resolve(relativePath);
   }
 
+  private boolean checkImageExists(String imageName) {
+    final String[] split = imageName.split(":");
+    final String basename = split[0];
+    final String tag = split.length > 1 ? split[1] : "";
+
+    try {
+      final Process checkProcess = new ProcessBuilder("./image_exists.sh", basename, tag).start();
+      checkProcess.waitFor();
+      return checkProcess.exitValue() == 0;
+
+    } catch (IOException | InterruptedException e) {
```

Comment:
i'm catching exceptions because there are no exceptions in the iface. i don't want to add them because i don't know that they really make sense in this iface. 

this is party of the reason i'm hesitating as to whether this code belongs here.